### ISSUE TITLE
ComposeRedundantComposable false positive when invoking a @Composable function-typed value held in a property

### ISSUE THREAD

**Issue by UniqueEVE** [slackhq/compose-lints#568]
`ComposeRedundantComposable` incorrectly flags a `@Composable` function as redundant when its only use of the composition is invoking a value whose type is a `@Composable` function type (e.g. a `@Composable () -> Unit` stored in a property/field), rather than calling a `@Composable`-annotated function or reading a `@Composable` property.

  ### To Reproduce:

  ```kotlin
  sealed interface MessageViewState {
      data object Hidden : MessageViewState
      class Showing(val composable: @Composable () -> Unit) : MessageViewState
  }

  @Composable
  fun MessageDisplay(state: MessageViewState) {
      when (state) {
          MessageViewState.Hidden -> Unit
          is MessageViewState.Showing -> state.composable() 
      }
  }
```
Run on `1.5.2` version.
